playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml

```diff
@@ -0,0 +1,133 @@
+  include_role:
+    name: golang-installation
+  vars:
+    go_tarball: "{{ golang_tarball_logging_e2e }}"
+    golang_path: "/usr/local"
+
+- name: Clone e2e tests repo
+  git:
+    repo: "https://{{ githubuser | urlencode }}:{{ githubpassword | urlencode }}@github.com/openshift/openshift-tests-private.git"
+    dest: "{{ test_dir }}"
+    version: "{{ ocp_logging_tests_private_e2e_git_branch }}"
+
+- name: Run make build command at target
+  shell: make
+  environment: "{{ logging_e2e_env }}"
+  args:
+    chdir: "{{ test_dir }}"
+
+- name: Check if the binary is created
+  shell: ls -hl {{ test_dir }}/bin/extended-platform-tests | wc -l
+  args:
+    chdir: "{{ test_dir }}"
+  register: bin_output
+  failed_when: bin_output.stdout|int != 1
+
+- name: Copy openshift-tests-private binary
+  copy:
+    src: "{{ bin_path }}"
+    remote_src: true
+    dest: "/usr/local/bin/"
+    mode: '0755'
+
+- name: Create results directory
+  file:
+    path: "{{ results_dir }}"
+    state: "{{ item }}"
+    mode: '0755'
+  with_items:
+  - absent
+  - directory
+
+# Check Nodes status
+- name: Check all nodes are Ready
+  shell: oc wait --all --for=condition=Ready nodes --timeout=300s
+
+# Check CO status
+- name: Check if all co are in AVAILABLE state and check no any co are in PROGRESSING state and DEGRADED state
+  shell: oc get co -n default --no-headers | awk '{ print $3 $4 $5 }' | grep -w -v TrueFalseFalse | wc -l
+  register: co_status
+  until: co_status.stdout|int == 0
+  retries: 30
+  delay: 120
+  ignore_errors: true
+
+- name: Fail if Cluster operators are not in good status
+  fail:
+    msg: "Cluster operators are not in good status. Skipping E2E tests !!"
+  when: 0 != co_status.stdout|int
+
+# Check PODS
+- name: Verify all PODS are Running or Completed
+  shell: oc get pods -A | awk 'NR>1 {if($4!="Running" && $4!="Completed" ){print $2" "$4}}'
+  register: pods_status
+  until: pods_status.stdout|length == 0
+  retries: 30
+  delay: 60
+  ignore_errors: true
+
+- debug:
+    msg: "Unhealthy PODS: {{ pods_status.stdout_lines }}"
+  when: 0 != pods_status.stdout|length
+
+
+# Running E2E
+- name: Prepare test suites and run e2e tests
+  shell: |
+    ./bin/extended-platform-tests run all --dry-run | grep "logging" > logging_tests.txt
+    ./bin/extended-platform-tests run -f ./logging_tests.txt {{ repository_command }} -o {{ results_dir }}/logging_tests_output.txt
+    sed -e 's/\"/\\"/g;s/.*/\"&\"/'  {{ results_dir }}/logging_tests_private_e2e_output.txt   | awk '/Failing tests:/,EOF' | tail -n +3 | head -n -4 > {{ results_dir }}/failed-logging-tests-private-e2e-results.txt
+  args:
+    chdir: "{{ test_dir }}"
+  environment: "{{ logging_e2e_env }}"
+  when: logging_e2e_testcases == "all" or logging_e2e_testcases == ""
+  register: opp
+
+- name: Prepare test suites and run critical e2e tests
+  shell: |
+    ./bin/extended-platform-tests run all --dry-run | grep -E "sig-openshift-logging.*Critical" > logging_tests_critical.txt
```

Comment:
Critical test cases should also be included in the logging tests, right? Are these redundant ?

Reply:
Yes, here we can choose between Critical logging testcases or all the logging related testcases. Critical testcases will be part of all logging testcases.

Reply:
ok.